### PR TITLE
test: sandbox deletion

### PR DESCRIPTION
**Pull request by rgrinberg**
This test demosntrates that errors from deleting the sandbox do not
shadow rule errors.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Comment by Alizter** [comment]
I've triggered this when making cram tests that remove read permissions from files.

**Comment by Alizter** [comment]
@rgrinberg Doesn't look like `-R` is in posix `chmod`.

File: test/blackbox-tests/test-cases/interfere-sandbox-deletion.t [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Create a bad rule that doesn't allow the sandbox to be deleted.
+
+Note that this rule also fails. We make sure that we see both the error from
+the rule and the sandbox cleanup.
+
+  $ cat >dune <<EOF
+  > (rule
+  >  (target foo)
+  >  (action
+  >   (with-accepted-exit-codes 1
```
Comment (snowleopard) [inline test/blackbox-tests/test-cases/interfere-sandbox-deletion.t:10]: Why? Was this command supposed to fail too? (It doesn't appear to, at the moment.)

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/interfere-sandbox-deletion.t:10]: Yes, the intention is for the command to produce the target _and_ fail. It also does fail, see `Command exited with code 0.` in the error message.

**Review by Alizter — APPROVED** [review]

File: test/blackbox-tests/test-cases/interfere-sandbox-deletion.t (hunk 1 above)
Comment (snowleopard) [reply test/blackbox-tests/test-cases/interfere-sandbox-deletion.t:10]: Oh, I see. Why not `exit 1` instead? Seems way simpler.

Comment (snowleopard) [reply test/blackbox-tests/test-cases/interfere-sandbox-deletion.t:10]: (And maybe `echo` something to confirm the error gets captured and displayed.)

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/interfere-sandbox-deletion.t]: Should be good now.

**Review by snowleopard — APPROVED** [review]
Thank! Looks great now.

**Pull request merged**